csharp/ql/src/experimental/CWE-095/razor_parse.ql

```diff
@@ -0,0 +1,71 @@
+/**
+ * @name Server-Side Template Injection in RazorEngine
+ * @description User-controlled data may be evaluated, leading to arbitrary code execution.
+ * @kind path-problem
+ * @problem.severity error
+ * @precision high
+ * @id csharp/razor-injection
+ * @tags security
+ */
+
+import csharp
+import semmle.code.csharp.dataflow.TaintTracking
+import DataFlow::PathGraph
+
+/*
+ * The offending method is Parse from RazorEngine.Razor.
+ */
+
+class RazorEngineClass extends Class {
+  RazorEngineClass() { this.hasQualifiedName("RazorEngine.Razor") }
+
+  Method getIsValidMethod() {
```

Comment:
Why is this called `getIsValidMethod` and not e.g. `get(A)ParseMethod`?

Reply:
Correct. Old function name. Changed it to getParseMethod()